### ISSUE TITLE
False positive for `ComposeUnstableReceiver` starting with AGP 8.5.0-alpha04 and K2 UAST

### ISSUE THREAD

**Issue by SimonMarquis** [slackhq/compose-lints#326]
The following code will trigger [`ComposeUnstableReceiver`](https://slackhq.github.io/compose-lints/rules/#unstable-receivers) when running with this configuration:

- Kotlin 2.0.0-RC1
- K2 UAST (`android.lint.useK2Uast=true`)
- Lint 8.5.0-alpha04 and above (`-Pandroid.experimental.lint.version=8.5.0-alpha04`) (worked fine on `alpha03`)

```kotlin
import androidx.compose.ui.graphics.Color

public val Color.dim1: Color
    @Composable get() = this.copy(alpha = MyTheme.colors.dim1)
```

**Comment by ZacSweers** [comment]
Thanks for the report, we'll take a look. I did notice that alpha04 seemed to have a better understanding of some kotlin language features and being able to see through syntax better
